lib/em/protocols/linetext2.rb

```diff
@@ -52,45 +52,51 @@ def receive_data data
         @lt2_delimiter ||= "\n"
         @lt2_linebuffer ||= []
 
-        if @lt2_mode == :lines
-          if ix = data.index( @lt2_delimiter )
-            @lt2_linebuffer << data[0...ix]
-            ln = @lt2_linebuffer.join
-            @lt2_linebuffer.clear
-            if @lt2_delimiter == "\n"
-              ln.chomp!
+        remaining_data = data
+
+        while remaining_data.length > 0
```

Comment:
I guess using length mirrors the use of [] to select ranges of characters to operate on